multi-buffer-search/src/services/changeTracker.ts

```
@@ -0,0 +1,103 @@
+      const originalContent = this.originalContent.get(multiLine)
+      if (originalContent !== undefined && currentContent !== originalContent) {
+        const fileKey = excerpt.fileUri.toString()
+        if (!changesByFile.has(fileKey)) {
+          changesByFile.set(fileKey, {
+            uri: excerpt.fileUri,
+            changes: []
+          })
+        }
+        const sourceLineOffset = multiLine - excerpt.multiBufferRange.start.line
+        const sourceLine = excerpt.sourceRange.start.line + sourceLineOffset
+        changesByFile.get(fileKey)!.changes.push({
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)